cli/http_util.rs

```diff
@@ -339,15 +345,24 @@ impl HttpClient {
   pub async fn get_redirected_response<U: reqwest::IntoUrl>(
     &self,
     url: U,
+    maybe_header: Option<(HeaderName, HeaderValue)>,
   ) -> Result<Response, AnyError> {
     let mut url = url.into_url()?;
-    let mut response = self.get_no_redirect(url.clone())?.send().await?;
+
+    let mut builder = self.get_no_redirect(url.clone())?;
+    if let Some((header_name, header_value)) = maybe_header.as_ref() {
+      builder = builder.header(header_name, header_value);
+    }
+    let mut response = builder.send().await?;
     let status = response.status();
     if status.is_redirection() {
       for _ in 0..5 {
         let new_url = resolve_redirect_from_response(&url, &response)?;
-        let new_response =
-          self.get_no_redirect(new_url.clone())?.send().await?;
+        let mut builder = self.get_no_redirect(new_url.clone())?;
+        if let Some((header_name, header_value)) = maybe_header.as_ref() {
+          builder = builder.header(header_name, header_value);
+        }
+        let new_response = builder.send().await?;
```

Comment:
You should only serve auth tokens after a redirect if the original and redirected URL have the same origin.